items/resources.py

```diff
@@ -166,30 +167,29 @@ class Meta:
     pass
 
   def prepend_urls(self):
-    return [
-      url(r'^registration/$', self.wrap_view('registration'), name='api_registration')
-    ]
+    return [url(r'^registration/$', self.wrap_view('registration'), name='api_registration')]
 
   def registration(self, request, **kwargs):
-    if request.method == 'POST':
-      try:
-        data = json.loads(request.body)
-        username = data['username']
-        password = data['password']
-        confirm_password = data['confirmPassword']
-        email = data['email']
-      except KeyError:
-        return HttpResponseServerError('Wrong params')
-      if password != confirm_password:
-        return HttpResponseServerError('Passwords are not equal')
-      try:
-        user = User.objects.create_user(username=username, password=password, email=email)
-      except IntegrityError:
-        return HttpResponseServerError('There is user with this name')
-      response = self.create_response(request, Unpacking.unpack_user(user))
-      return HttpResponse(response)
-    else:
-      return HttpResponseServerError('Forbidden method')
+    self.method_check(request, allowed=['post'])
+    data = self.deserialize(request, request.body,
+    format=request.META.get('CONTENT_TYPE', 'application/json'))
+    username = data.get('username')
+    password = data.get('password')
+    repeatPassword = data.get('confirmPassword')
+    try:
+      user=User.objects.get(username=username)
+      return self.create_response(request, { 'reason' : 'userexist' }, HttpUnauthorized)
+    except ObjectDoesNotExist:
+      if password == repeatPassword:
+        email = data.get('email')
+        user = User.objects.create_user(username, email, password)
+        user.save()
```

Comment:
https://docs.djangoproject.com/en/1.8/ref/contrib/auth/#django.contrib.auth.models.UserManager.create_user
`create_user` creates, saves and returns user